add RasterVectorJoin Operator (#47)
geo-engine/docs: src/operators/rastervectorjoin.md

# RasterVectorJoin

The `RasterVectorJoin` operator allows combining a single vector input and multiple raster inputs.
For each raster input, a new column is added to the collection from the vector input.
The new column contains the value of the raster at the location of the vector feature.
For features covering multiple pixels like `MultiPoints` or `MultiPolygons`, the value is calculated using an aggregation function selected by the user.
The same is true if the temporal extend of a vector feature covers multiple raster time-steps.
More details are described below.

**Example**:

> You have a collection of aggricaltural fields (`Polygons`) and a collection of raster images containing the monthly NDVI value for each pixel.

Reviewer: ```suggestion
You have a collection of agricultural fields (`Polygons`) and a collection of raster images containing each pixel's monthly NDVI value.
```

typo + grammarly suggestion

> The `RasterVectorJoin` operator allows you to do this.

Reviewer: Clarify: to do what?

Author: [ec1758e](https://github.com/geo-engine/docs/pull/47/commits/ec1758eeb071d2f3f612b9b54b9051f980628b87)

You now have multiple options to calculate the NDVI value for each field.

> For a quick evaluation, you can use the `first` aggregation function.

Reviewer: Maybe first state what it is good for in general.
Then say, that it is cheap and also good for exploratory use.

| --------------------- | ------------------------- | ----------------------------------------------------------------------------------- | ---------------------------------- |
| `names`               | `Array<String>`           | Each name reflects the output column of the join result.                            | <pre>"["NDVI", "Elevation"]"</pre> |
| `featureAggregation`  | `first` or `mean`         | The aggregation function to use for features covering multiple pixels.              | <pre>"first"</pre>                 |
| `temporalAggregation` | `none`, `first` or `mean` | The aggregation function to use for features covering multiple (raster) time steps. | <pre>"none"</pre>                  |

## Errors

If the lenght of `names` is not equal to the number of raster inputs, an error is thrown.

## Example JSON

```json
{
  "type": "RasterVectorJoin",
  "params": {
    "names": ["NDVI"],
    "featureAggregation": "first",
    "temporalAggregation": "mean"
  },
  "sources": {
    "vector": {
      "type": "OgrSource",
      "params": {

>         "data": {
>           "type": "internal",
>           "datasetId": "e977b123-ca47-4c5b-aace-481119826aaf"
>         }

Reviewer: ```suggestion
        "data": "ne_10m_ports"
```

      }
    },
    "rasters": [
      {
        "type": "GdalSource",
        "params": {

>           "data": {
>             "type": "internal",
>             "datasetId": "b6191257-6d61-4c6b-90a4-ebfb1b23899d"
>           }

Reviewer: ```suggestion
          "data": "ndvi"
```
